added RoleOnly option for iamserviceaccounts (#3122)
weaveworks/eksctl: userdocs/src/usage/iamserviceaccounts.md

eksctl create iamserviceaccount --cluster=<clusterName> --name=<serviceAccountName> --role-name "custom-role-name"
```

When the service account is created and managed by some other tool, such as helm, use `--role-only` to prevent conflicts.

> The other toll is then responsible for adding the role ARN annotation.

Reviewer: ```suggestion
The other tool is then responsible for maintaining the role ARN annotation.
```

```console

> eksctl create iamserviceaccount --cluster=<clusterName> --name=<serviceAccountName> --role-only --role-name "custom-role-name"

Reviewer 2: Minor nit, just to be consistent:

```suggestion
eksctl create iamserviceaccount --cluster=<clusterName> --name=<serviceAccountName> --role-only --role-name=<customRoleName>
```

```

Currently, to update a role you will need to re-create, run `eksctl delete iamserviceaccount` followed by `eksctl create iamserviceaccount` to achieve that.

!!!note

>     `eksctl delete iamserviceaccount` deletes service accounts even if they were not created by `eksctl create iamserviceaccount`.

Reviewer: ```suggestion
    `eksctl delete iamserviceaccount` deletes Kubernetes `ServiceAccounts` even if they were not created by `eksctl`.
```
